File: public/examples/lua/basic/04_control_flow.lua
-- Lua Basic: Control Flow

local score = 85

if score >= 90 then
    print("Grade: A")
elseif score >= 80 then
    print("Grade: B")
else
    print("Grade: C")
end

-- While loop
local count = 3
while count > 0 do
    print("Counting:", count)
    count = count - 1
end

-- For loop (start, end, step)
print("Range for:")
for i = 1, 3 do
    print(i)
end

File: public/examples/rust/basic/04_control_flow.rs
// Rust (Rune) Basic: Control Flow

pub fn main() {
    let score = 85;

    if score >= 90 {
        println!("Grade: A");
    } else if score >= 80 {
        println!("Grade: B");
    } else {
        println!("Grade: C");
    }

    println!("Looping:");
    for i in 0..3 {
        println!("i: {}", i);
    }
}
